src/hooks/useDebounce.js

```diff
@@ -0,0 +1,42 @@
+import { useState, useEffect, useCallback } from 'react'
+import debounce from 'lodash.debounce'
+
+/**
+ * Custom hook to debounce a function for a specified amount of time
+ */
+const useDebounce = (fn, debounceDurationMs) => {
```

Comment:
You should jsdoc comment these if you can't use typesecript, specifically to explain how `0` disables debouncing. It should probably also be documented at the button level, esp. considering debouncing will be the new default behavior.

Reply:
updated https://github.com/getethos/ethos-design-system/pull/381/commits/5a3cd0f8eb8ec03f0f4ecaaa45978ae721ac4631